src/darwin/Framework/CHIP/MTRDeviceController_XPC.mm

```diff
@@ -72,28 +72,28 @@ @implementation MTRDeviceController_XPC
 
 - (void)_updateRegistrationInfo
 {
-    dispatch_async(self.workQueue, ^{
-        std::lock_guard lock(*self.deviceMapLock);
+    os_unfair_lock_lock(self.deviceMapLock);
```

Comment:
Why is this not using std::lock_guard?

You do need to import MTRUnfairLock.h to make that compile, though?